### PR TITLE
Check for GLEW library on build

### PR DESCRIPTION
**Pull request by GArik**
GLEW is used by rgbd module in OpenCV contrib, so check if it is available.

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
build_image:Custom=qt:16.04
buildworker:Custom=linux-1
```

### CODE REVIEW

**Comment by GArik** [comment]
Required for: https://github.com/opencv/opencv_contrib/pull/2731

File: cmake/OpenCVUtils.cmake [hunk 1]
```diff
@@ -808,7 +808,7 @@ macro(ocv_check_modules define)
     pkg_check_modules(${define} ${ARGN})
   endif()
   if(${define}_FOUND)
-    set(HAVE_${define} 1)
+    set(HAVE_${define} TRUE)
```
Comment (alalek) [inline cmake/OpenCVUtils.cmake:811]: Why?

File: cmake/OpenCVFindLibsGUI.cmake [hunk 2]
```diff
@@ -82,6 +82,7 @@ if(WITH_OPENGL)
       else()
         ocv_include_directories(${OPENGL_INCLUDE_DIR})
       endif()
+      ocv_check_modules(GLEW glew)
```
Comment (alalek) [inline cmake/OpenCVFindLibsGUI.cmake:85]: 1. This doesn't work on native Win32 (utilizes `pkg-config`)
2. Use CMake's way instead: https://cmake.org/cmake/help/v3.0/module/FindGLEW.html

File: cmake/OpenCVUtils.cmake (hunk 1 above)
Comment (GArik) [reply cmake/OpenCVUtils.cmake:811]: I'll revert this change, it doesn't make any difference, a boolean value just looks clearer to me.

File: cmake/OpenCVFindLibsGUI.cmake (hunk 2 above)
Comment (GArik) [reply cmake/OpenCVFindLibsGUI.cmake:85]: FindGLEW doesn't work on Linux -- it finds only GLEW library; pkg-config returns all necessary flags:
```
$ pkg-config --libs glew
-lGLEW -lX11 -lGLU -lGL
```

**Pull request closed without merging**